src/main/java/sorald/processor/StringLiteralInsideEqualsProcessor.java

```diff
@@ -0,0 +1,77 @@
+package sorald.processor;
+
+import sorald.annotations.ProcessorAnnotation;
+import spoon.reflect.code.BinaryOperatorKind;
+import spoon.reflect.code.CtBinaryOperator;
+import spoon.reflect.code.CtExpression;
+import spoon.reflect.code.CtInvocation;
+import spoon.reflect.declaration.CtElement;
+import spoon.reflect.declaration.CtMethod;
+import spoon.reflect.declaration.CtType;
+import spoon.reflect.reference.CtExecutableReference;
+
+@ProcessorAnnotation(
+        key = 1132,
+        description =
+                "Strings literals should be placed on the left side when checking for equality")
+public class StringLiteralInsideEqualsProcessor extends SoraldAbstractProcessor<CtInvocation<?>> {
+
+    @Override
+    protected void repairInternal(CtInvocation<?> element) {
+        if (element.getExecutable().getSignature().equals("equals(java.lang.Object)")
+                || element.getExecutable()
+                        .getSignature()
+                        .equals("equalsIgnoreCase(java.lang.String)")) {
+            CtType stringClass = getFactory().Class().get(String.class);
+            CtMethod ctMethodToBeCalled = null;
+            if (element.getExecutable().getSignature().equals("equals(java.lang.Object)")) {
+                ctMethodToBeCalled = (CtMethod) stringClass.getMethodsByName("equals").get(0);
+            } else {
+                ctMethodToBeCalled =
+                        (CtMethod) stringClass.getMethodsByName("equalsIgnoreCase").get(0);
+            }
+            CtExecutableReference ctExecutableReferenceToMethodToBeCalled =
+                    getFactory().Executable().createReference(ctMethodToBeCalled);
+
+            CtInvocation newInvocation =
+                    getFactory()
+                            .Code()
+                            .createInvocation(
+                                    element.getArguments().get(0),
+                                    ctExecutableReferenceToMethodToBeCalled,
+                                    element.getTarget());
+
+            element.replace(newInvocation);
+
+            // the following is to handle the case in which there is a null check on the variable
+            // used as target
+            CtExpression<?> variable = (CtExpression<?>) newInvocation.getArguments().get(0);
```

Comment:
Cast is redundant if `newInvocation` is properly parameterized.

```suggestion
            CtExpression<?> variable = newInvocation.getArguments().get(0);
```

Reply:
Changed.